### PR TITLE
Acquire password in cli when its stdout is a pipe

### LINKED ISSUE #916: presto-cli should acquire password using stdin&stderr when stdout is a pipe
This should be possible:
```
presto-cli --user me --password --execute 'select 1' > /dev/null
```

Currently, this fails with
```
Exception in thread "main" java.lang.RuntimeException: No console from which to read password
	at io.prestosql.cli.Console.getPassword(Console.java:173)
	at io.prestosql.cli.Console.run(Console.java:136)
	at io.prestosql.cli.Presto.main(Presto.java:31)
```

It should work just the same way as `sudo echo x > /tmp/x` does.

### PR DESCRIPTION
**Pull request by Praveen2112**
Fixes #916

### CODE REVIEW

File: presto-cli/src/main/java/io/prestosql/cli/Console.java [hunk 1]
```diff
@@ -170,7 +170,13 @@ private String getPassword()
 
         java.io.Console console = System.console();
         if (console == null) {
-            throw new RuntimeException("No console from which to read password");
+            try {
+                jline.console.ConsoleReader consoleReader = new jline.console.ConsoleReader(System.in, System.err);
```
Comment (electrum) [inline presto-cli/src/main/java/io/prestosql/cli/Console.java]: Why does `ConsoleReader` need to be fully qualified? I don't see any conflicting class names.

File: presto-cli/src/main/java/io/prestosql/cli/Console.java [hunk 2]
```diff
@@ -170,7 +170,13 @@ private String getPassword()
...
+                return consoleReader.readLine("Password: ", Character.valueOf((char) 0));
```
Comment (electrum) [inline presto-cli/src/main/java/io/prestosql/cli/Console.java]: This should work without the explicit boxing to `Character`:
```java
return consoleReader.readLine("Password: ", (char) 0);
```

File: presto-cli/src/main/java/io/prestosql/cli/Console.java [hunk 3]
```diff
@@ -170,7 +170,13 @@ private String getPassword()
...
+            }
+            catch (IOException exception) {
+                throw new RuntimeException("No console from which to read password");
```
Comment (electrum) [inline presto-cli/src/main/java/io/prestosql/cli/Console.java]: Throw as `UncheckedIOException`. Also, this error might not be correct anymore. Probably better to make it more general: `Failed to read password from console`

File: presto-cli/src/main/java/io/prestosql/cli/Console.java [hunk 4]
```diff
@@ -170,7 +170,13 @@ private String getPassword()
 
         java.io.Console console = System.console();
         if (console == null) {
```
Comment (electrum) [inline presto-cli/src/main/java/io/prestosql/cli/Console.java]: Let's invert this so we aren't splitting the logic
```java
if (console != null) {
    ... read from console
}

// fallback to jline
...

File: presto-cli/src/main/java/io/prestosql/cli/Console.java (hunk 3 above)
Comment (Praveen2112) [reply presto-cli/src/main/java/io/prestosql/cli/Console.java]: Applied

File: presto-cli/src/main/java/io/prestosql/cli/Console.java (hunk 2 above)
Comment (Praveen2112) [reply presto-cli/src/main/java/io/prestosql/cli/Console.java]: Applied

File: presto-cli/src/main/java/io/prestosql/cli/Console.java (hunk 1 above)
Comment (Praveen2112) [reply presto-cli/src/main/java/io/prestosql/cli/Console.java]: Applied

**Comment by Praveen2112** [comment]
@electrum Thanks for your insights. Have applied your comments.

**Pull request merged**